src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final double radius;
+    private final ColorSupplier color;
```

Comment:
The use of ColorSupplier as a field in Circle is not appropriate. You should pass a Color object to the constructor instead of a ColorSupplier. The responsibility of ColorSupplier is to supply a color, not to be stored as a property of a figure.